cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexManagerImpl.java

```diff
@@ -256,6 +260,32 @@ public String ensureIndexed(List<FieldSort> fieldNames,
                                 String indexName,
                                 IndexType indexType,
                                 String tokenize) throws QueryException {
+
+        // If the index name is null, a name should be generated, but first we should check
+        // if an index already exists that matches the request index definition.
+        if (indexName == null) {
+            List<Index> indexes = this.listIndexes();
+            Set<FieldSort> fieldNamesSet = new HashSet<FieldSort>(filterMeta(fieldNames));
+            for (Index index : indexes) {
+                Set<FieldSort> indexFieldNamesSet = new HashSet<FieldSort>(filterMeta(index.fieldNames));
+                if (!(indexFieldNamesSet.containsAll(fieldNamesSet) && indexFieldNamesSet.size() == fieldNamesSet.size())) {
```

Comment:
Actually, scratch that last comment, shouldn't this just be `if (!indexFieldNamesSet.equals(fieldNamesSet))` as these are both sets.